File: julia/clock/clock.jl
import Base.+
import Base.-
import Base.show
using Printf

struct clock
    hours::Int
    minutes::Int
end

function Clock(h::Int, m::Int)
    while m < 0
        m += 60
        h -= 1
    end
    while h < 0
        h += 24
    end
    clock((h + (m ÷ 60)) % 24, m %= 60)
end

(+)(x::clock, y::Dates.Minute) = Clock(x.hours, x.minutes + y.value)

(-)(x::clock, y::Dates.Minute) = Clock(x.hours, x.minutes - y.value)

(show)(io::IO, c::clock) = print(io, @sprintf("\"%02d:%02d\"", c.hours, c.minutes))

File: csharp/clock/Clock.cs
using System;

public class Clock
{
    private int _hours;
    private int _mins;
    
    public Clock(int hours, int minutes)
    {
        _hours = hours;
        _mins = minutes;
        CleanOverflow();
    }

    public Clock Add(int minutesToAdd) => new(_hours, _mins + minutesToAdd);

    public Clock Subtract(int minutesToSubtract) => Add(-minutesToSubtract);

    public override string ToString() => $"{_hours:d2}:{_mins:d2}";

    public override bool Equals(Object obj)
    {
        Clock other = (Clock)obj;
        return other != null && (this._hours == other._hours) && (this._mins == other._mins);
    }

    private void CleanOverflow()
    {
        int mins = _mins;
        while (mins < 0)
        {
            mins += 60;
            _hours--;
        }
        while (_hours < 0) _hours += 24;
        _mins = mins % 60;
        _hours = (_hours + (mins / 60)) % 24;
    }
}
